### PR TITLE
add: postprocessors

### PR DESCRIPTION
**Pull request by iwilltry42**
Similar to ingestion flow transformers, but acting on the returned documents and with additional query context.

### CODE REVIEW

File: pkg/datastore/postprocessors/content.go
Comment (iwilltry42) [inline pkg/datastore/postprocessors/content.go:1]: Example: Does the content include any explicit words?

File: pkg/datastore/postprocessors/postprocessors.go [hunk 1]
```diff
@@ -2,27 +2,46 @@
...
+
+func (t *TransformerWrapper) Transform(ctx context.Context, query string, docs []vs.Document) ([]vs.Document, error) {
+	return t.DocumentTransformer.Transform(ctx, docs)
+}
 
-var PostprocessorMap = map[string]Postprocessor{}
+var PostprocessorMap = map[string]Postprocessor{
+	"extra_metadata":                  NewTransformerWrapper(&transformers.ExtraMetadata{}),
+	"keywords":                        NewTransformerWrapper(&transformers.KeywordExtractor{}),
+	"filter_markdown_docs_no_content": NewTransformerWrapper(&transformers.FilterMarkdownDocsNoContent{}),
```
Comment (iwilltry42) [inline pkg/datastore/postprocessors/postprocessors.go:33]: Input Transformers can easily be used as output postprocessors

File: pkg/datastore/postprocessors/similarity.go
Comment (iwilltry42) [inline pkg/datastore/postprocessors/similarity.go:1]: Example: only return documents with a similarity score >= 0.9

**Review by StrongMonkey — APPROVED** [review]
Will user be able to define such postProcessor in config file?

**Comment by iwilltry42** [comment]
@StrongMonkey yep: https://github.com/gptscript-ai/knowledge/blob/main/pkg%2Fflows%2Fconfig%2Fconfig.go#L52

**Pull request merged**